README.rst

```diff
@@ -425,9 +471,27 @@ Constructors for Sorted Containers
   the parameters of the type.  Ordering argument ``o`` is
   optional and defaults to ``Forward``.
 
+``SortedMultiDict(k,v,o)``
+  Construct a SortedMultiDict using keys given by ``k``, values
+  given by ``v`` and ordering object ``o``.  The ordering object
+  defaults to ``Forward`` if not specified.  The two arguments
+  ``k`` and ``v`` are 1-dimensional arrays of the same length in 
+  which ``k`` holds keys and ``v`` holds the corresponding values.
+  To construct an empty SortedMultiDict with given types ``K`` and
+  ``V``, use
+  ``SortedMultiDict(K[], V[], o)``.
```

Comment:
@kmsquire and @vtjnash,

I looked at refpointer.jl, and here is my interpretation of the purpose of this code.  The idea is to be able to pass an immutable object, say x, a Float64, to a function f and yet the function can still mutate it.  This is accomplished as follows.  If x is bare Float64, then it is wrapped in a type.  If x is an element of an array a at position i, then (a,i) are passed to f.  However, f does not need to know what is the source of x since its formal argument can be Ref{Float64}, which covers both cases.

I can certainly extend this to the three sorted containers (@kmsqure: I'm not sure this is directly related to the deref operation in my code, but maybe you can explain more) I have a couple of questions about this construction.

(1) Is there a plan to extend this to associative?  In other words, if D is of type Dict{ASCIIString,Float} and k is a string (key), then one could say x = RefDict(D,k) to get another Ref{Float} object?  If so, is the extension to associative already written?

(2) You are probably already aware of this, but there is a performance issue with this construct.  Specifically, if one says x = Ref(a,i) where a is an array and i is a subscript, then the new object x is created on the heap rather than the stack (even though internally it consists of just a pointer and an integer), so you pay a price to use this in an inner loop.